tests/dragonfly/search_test.py

```diff
@@ -558,3 +562,241 @@ def make_car(producer, description, speed):
+                    break
+                except Exception:
+                    # Expected during index operations
+                    time.sleep(0.001)
+        finally:
+            client.close()
+
+    def reindexer_thread():
+        """Thread that drops and recreates index"""
+        client = redis.Redis(port=df_server.port, decode_responses=True)
+        try:
+            while not stop_threads.is_set() and not server_crashed.is_set():
+                try:
+                    # Drop index
+                    client.execute_command("FT.DROPINDEX", index_name)
+
+                    # Recreate index
+                    client.execute_command(
+                        "FT.CREATE",
+                        index_name,
+                        "ON",
+                        "HASH",
+                        "PREFIX",
+                        "1",
+                        "doc:",
+                        "SCHEMA",
+                        "content",
+                        "TEXT",
+                        "SORTABLE",
+                        "tags",
+                        "TAG",
+                        "SORTABLE",
+                        "numeric_field",
+                        "NUMERIC",
+                        "SORTABLE",
+                    )
+                    # Small delay between index operations
+                    time.sleep(0.01)
+                except (redis.ConnectionError, OSError):
+                    server_crashed.set()
+                    break
+                except Exception:
+                    # Expected during concurrent operations
+                    time.sleep(0.01)
+        finally:
+            client.close()
+
+    def searcher_thread():
+        """Thread that performs search operations"""
+        client = redis.Redis(port=df_server.port, decode_responses=True)
+        try:
+            while not stop_threads.is_set() and not server_crashed.is_set():
+                try:
+                    random_val = random.randint(1, 20000)
+                    random_tag_val = random.randint(0, 20)
+
+                    # Various search operations
+                    client.execute_command("FT.SEARCH", index_name, f"@content:item{random_val}")
+
+                    client.execute_command(
+                        "FT.SEARCH", index_name, f"@tags:{{tagA|tagC|tag{random_tag_val}}}"
+                    )
+
+                    client.execute_command(
+                        "FT.SEARCH", index_name, f"@numeric_field:[{random_val} {random_val + 100}]"
+                    )
+
+                    # Small delay between search operations
+                    time.sleep(0.005)
+
+                except (redis.ConnectionError, OSError):
+                    server_crashed.set()
+                    break
+                except Exception:
+                    # Expected during index operations
+                    time.sleep(0.005)
+        finally:
+            client.close()
+
+    def writer2_thread():
```

Comment:
you do not need to clone the same function twice, you can just launch a second thread with the same function.